app/presenters/collaborator_presenter.rb

```diff
@@ -6,27 +6,77 @@ def initialize(seller:, collaborator: nil)
     @collaborator = collaborator
   end
 
+  DEFAULT_PERCENT_COMMISSION = 50
+  MAX_PRODUCTS_WITH_AFFILIATES_TO_SHOW = 10
+
   def new_collaborator_props
-    {
-      products: all_products,
-      collaborators_disabled_reason: seller.has_brazilian_stripe_connect_account? ? "Collaborators with Brazilian Stripe accounts are not supported." : nil,
-    }
+    collaborator_form_props.compact
   end
 
   def collaborator_props
     collaborator.as_json.merge(products:)
   end
 
   def edit_collaborator_props
-    collaborator&.as_json&.merge({
-                                   products: all_products,
-                                   collaborators_disabled_reason: seller.has_brazilian_stripe_connect_account? ? "Collaborators with Brazilian Stripe accounts are not supported." : nil,
-                                 })
+    collaborator_form_props.compact
+  end
+
+  def inertia_shared_props
+    {
+      collaborators_disabled_reason: seller.has_brazilian_stripe_connect_account? ? "Collaborators with Brazilian Stripe accounts are not supported." : nil,
+    }
+  end
+
+  def incoming_collaborator_props(collaborator)
```

Comment:
This was  moved from the  `Api::Internal::Collaborators::IncomingsController`

Link - https://github.com/antiwork/gumroad/pull/2537/changes#diff-e7be6356df4c82ac60946f05a213ed5a19ff1794c12c598c908c1c2ed0d91aa0L13